### ISSUE TITLE
Get ind_matrix yields MemoryReport

### ISSUE THREAD

**Issue by proskurin** [hudson-and-thames/mlfinlab#130]
For datasets > 5000 labels get_ind_matrix yields memory error (which is explainable 5000x150000 can't fit into memory). Consider using scipy.sparse?

**Comment by imcu** [comment]
How about considering Dask dataframe https://docs.dask.org/en/latest/dataframe.html ?
It will be sharded across many tasks that individually will fit in memory.

**Comment by SergejGorev** [comment]
Got same issue recently, it requires unreasonably amount of memory. Depends on the dateset of course, but in my case `ind.shape=(492001, 448529)` I needed more then 1TiB. I could significantly reduce RAM requirements by changing dtype from `dtype="float64"` to  `dtype="uint8"`.
However, i still needed more then 250 GB RAM after that.

I tried to work with `scipy.sparse` but jit compiler does not accept scipy objects. BTW `scipy.sparse` needs less then 10 GB RAM for that dataset, however, it comes with a speed cost.

So, I searched on the interent and could "obviously" not find anything reasonable.

Dask DataFrame sounds like a possible solution @imcu 
any ideas how it implement it?
